demos/usage-remove-selected-elements.py

```diff
@@ -0,0 +1,138 @@
+import random
+
+import dash
+from dash.dependencies import Input, Output, State
+import dash_core_components as dcc
+import dash_html_components as html
+
+import dash_cytoscape as cyto
+
+asset_path = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    '..', 'assets'
+)
+
+app = dash.Dash(__name__, assets_folder=asset_path)
+server = app.server
+
+random.seed(2019)
+
+nodes = [
+    {'data': {'id': str(i), 'label': 'Node {}'.format(i)}}
+    for i in range(1, 21)
+]
+
+edges = [
+    {'data': {'source': str(random.randint(1, 20)), 'target': str(random.randint(1, 20))}}
+    for _ in range(30)
+]
+
+default_elements = nodes + edges
+
+styles = {
+    'json-output': {
+        'overflow-y': 'scroll',
+        'height': 'calc(50% - 25px)',
+        'border': 'thin lightgrey solid'
+    },
+    'tab': {'height': 'calc(98vh - 115px)'}
+}
+
+app.layout = html.Div([
+    html.Div(className='eight columns', children=[
+        cyto.Cytoscape(
+            id='cytoscape',
+            elements=default_elements,
+            layout={
+                'name': 'grid'
+            },
+            style={
+                'height': '95vh',
+                'width': '100%'
+            }
+        )
+    ]),
+
+    html.Div(className='four columns', children=[
+        dcc.Tabs(id='tabs', children=[
+            dcc.Tab(label='Actions', children=[
+                html.Button("Remove Selected Node", id='remove-button')
+            ]),
+            dcc.Tab(label='Tap Data', children=[
+                html.Div(style=styles['tab'], children=[
+                    html.P('Node Data JSON:'),
+                    html.Pre(
+                        id='tap-node-data-json-output',
+                        style=styles['json-output']
+                    ),
+                    html.P('Edge Data JSON:'),
+                    html.Pre(
+                        id='tap-edge-data-json-output',
+                        style=styles['json-output']
+                    )
+                ])
+            ]),
+
+            dcc.Tab(label='Selected Data', children=[
+                html.Div(style=styles['tab'], children=[
+                    html.P('Node Data JSON:'),
+                    html.Pre(
+                        id='selected-node-data-json-output',
+                        style=styles['json-output']
+                    ),
+                    html.P('Edge Data JSON:'),
+                    html.Pre(
+                        id='selected-edge-data-json-output',
+                        style=styles['json-output']
+                    )
+                ])
+            ])
+        ]),
+
+    ])
+])
+
+
+@app.callback(Output('cytoscape', 'elements'),
+              [Input('remove-button', 'n_clicks')],
+              [State('cytoscape', 'elements'),
+               State('cytoscape', 'selectedNodeData')])
+def remove_selected_nodes(_, elements, data):
+    if elements and data:
+        ids_to_remove = {ele_data['id'] for ele_data in data}
```

Comment:
Should this be a list instead?

Reply:
A set has a O(1) retrieval time compared to lists :)